pkg/query/querier.go

```diff
@@ -174,9 +174,7 @@ func (q *querier) Select(ms ...*labels.Matcher) (storage.SeriesSet, error) {
 	var (
 		mtx sync.Mutex
 		all []storepb.SeriesSet
-		// TODO(fabxc): errgroup will fail the whole query on the first encountered error.
-		// Add support for partial results/errors.
-		g errgroup.Group
+		wg  = &sync.WaitGroup{}
```

Comment:
The zero value of `sync.WaitGroup` is ready to use. No need for a heap instantiation.